src/repository/models.py

```diff
@@ -842,9 +854,46 @@ class VersionQueue(models.Model):
     date_decision = models.DateTimeField(blank=True, null=True)
     approved = models.BooleanField(default=False)
 
+    title = models.CharField(
+        max_length=300,
+        help_text=_('Your article title'),
+    )
+    abstract = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            'Please avoid pasting content from word processors as they can add '
+            'unwanted styling to the abstract. You can retype the abstract '
+            'here or copy and paste it into notepad/a plain text editor before '
+            'pasting here.',
+        )
+
+    )
+
     def approve(self):
         self.date_decision = timezone.now()
         self.approved = True
+
+        # Update the current version to have the Preprint's current title
+        # and abstract.
+        current_version = self.preprint.current_version
+        current_version.title = self.preprint.title
+        current_version.abstract = self.preprint.abstract
+
+        # Create a new PreprintVersion, this will now be the current_version.
+        PreprintVersion.objects.create(
+            preprint=self.preprint,
+            file=self.file,
+            version=self.preprint.next_version_number(),
+            moderated_version=self,
+        )
```

Comment:
Shouldn't we add the title and abstract when the `PreprintVersion` is created, instead of back-populating the previous one?

Reply:
I had thought of that but there are other ways a Title or Abstract can be changed to wanted it set at the point the version was no longer the `current_version`.